### PR TITLE
Boundary conditions module 

### LINKED ISSUE #25: Create `BoundaryConditions` module and interface
The following boundary conditions structs are included:
- [ ] `AbstractBoundaryCondition`
- [ ] `LoadBoundaryCondition`
- [ ] Tests `LoadsBoundaryCondition`
- [ ] `UserLoadBoundaryCondition`
- [ ] Tests `UserLoadBoundaryCondition`
- [ ] `SpringBoundaryCondition`
- [ ] Tests `SpringBoundaryCondition` 
- [ ] `DisplacementBoundaryCondition`
- [ ] Test `DisplacementBoundaryCondition`

### PR DESCRIPTION
**Pull request by mvanzulli**
Closes #25 

New structs has been included to add clamped and pinned supports. Moreover, new structs has been added to consider forces and moments.

### CODE REVIEW

File: src/interface/BoundaryConditions.jl [hunk 1]
```diff
@@ -23,23 +27,23 @@ abstract type AbstractBoundaryCondition end
...
-values(bc::ABC) = bc.values
+Base.values(bc::AbstractBoundaryCondition) = bc.values
 
 
 #####################################
 # Displacements Boundary Conditions #
 #####################################
 
-""" Displacement boundary condition struct.
+""" General displacement boundary condition struct.
```
Comment (jorgepz) [inline src/interface/BoundaryConditions.jl]:
```suggestion
""" Generalized displacement boundary condition struct.
```

Comment (mvanzulli) [reply src/interface/BoundaryConditions.jl]: Done

**Review by jorgepz — APPROVED** [review]
Vamo arruca
